### PR TITLE
hss 1.6 (new formula)

### PR DESCRIPTION
**Pull request by six-ddc**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/hss.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on "readline"
+
+  def install
+    system "make"
+    system "make", "install", "INSTALL_BIN=#{bin}"
+  end
+
+  test do
+    system "#{bin}/hss", "-v"
```
Comment (fxcoudert) [inline Formula/hss.rb]: This is unnecessary.

File: Formula/hss.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    system "#{bin}/hss", "-H", "127.0.0.1", "-u", "root", "command ls"
```
Comment (fxcoudert) [inline Formula/hss.rb]: Could we check the output to make sure that we get `Permission denied`? And possibly include a user other than `root` (e.g. `toto123`)

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/hss.rb (hunk 1 above)
Comment (six-ddc) [reply Formula/hss.rb]: get

File: Formula/hss.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class Hss < Formula
+  desc "Interactive parallel ssh client featuring autocomplete and asynchronous"
```
Comment (fxcoudert) [inline Formula/hss.rb]: `Interactive parallel ssh client` seems enough

File: Formula/hss.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  homepage "https://github.com/six-ddc/hss"
+  url "https://github.com/six-ddc/hss/archive/1.6.tar.gz"
+  sha256 "8516f3e24c9908f9c7ac02ee5247ce78f2a344e7fcca8a14081a92949db70049"
+  head "https://github.com/six-ddc/hss.git"
```
Comment (fxcoudert) [inline Formula/hss.rb]: Remove head please

File: Formula/hss.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  test do
+    begin
+      nc_read, nc_write = IO.pipe
+      nc_pid = fork do
+        exec "nc", "-4l", "9527", :out => nc_write
+      end
+      hss_read, hss_write = IO.pipe
+      hss_pid = fork do
+        exec "#{bin}/hss", "-H", "-p 9527 127.0.0.1", "-u", "root", "true", :out => hss_write
```
Comment (fxcoudert) [inline Formula/hss.rb]: Wrap this line to 80

File: Formula/hss.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  def install
+    system "make"
+    system "make", "install", "INSTALL_BIN=#{bin}"
+  end
+
+  test do
+    begin
+      nc_read, nc_write = IO.pipe
+      nc_pid = fork do
+        exec "nc", "-4l", "9527", :out => nc_write
```
Comment (fxcoudert) [inline Formula/hss.rb]: Let's not hardcode the port. You can get a valid port number this way:

```
    require "socket"

    server = TCPServer.new(0)
    port = server.addr[1]
    server.close
```

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by six-ddc** [comment]
PTAL @fxcoudert , Thanks

**Comment by fxcoudert** [comment]
Thanks @six-ddc for the pull request!

**Pull request closed without merging**